### PR TITLE
doc: crosslink {realize,resolve}GlobalName[NoOverload]?[WithInfo]?,

### PR DESCRIPTION
**Pull request by nomeata**

### CODE REVIEW

File: src/Lean/ResolveName.lean [hunk 1]
```diff
@@ -355,9 +359,13 @@ def ensureNonAmbiguous [Monad m] [MonadError m] (id : Syntax) (cs : List Name) :
   | [c] => pure c
   | _   => throwErrorAt id s!"ambiguous identifier '{id}', possible interpretations: {cs.map mkConst}"
 
-/-- Interpret the syntax `n` as an identifier for a global constant, and return a resolved
+/-- Interprets the syntax `n` as an identifier for a global constant, and return a resolved
 constant name. If there are multiple possible interpretations it will throw.
 
+Consider using `realizeGlobalConstNoOverload` if you need to handle reserved names, and
+`resolveGlobalConstNoOverloadWithInfo` resp. `realizeGlobalConstNoOverloadWithInfo` if you
```
Comment (david-christiansen) [inline src/Lean/ResolveName.lean:366]:
```suggestion
`realizeGlobalConstNoOverloadWithInfo` if you
```

The `resolve` version no longer exists.

File: src/Lean/ResolveName.lean [hunk 2]
```diff
@@ -355,9 +359,13 @@ def ensureNonAmbiguous [Monad m] [MonadError m] (id : Syntax) (cs : List Name) :
...
+want the syntax to have hover.
```
Comment (david-christiansen) [inline src/Lean/ResolveName.lean:367]:
```suggestion
want the syntax to show the resulting name's info on hover.
```

Comment (david-christiansen) [reply src/Lean/ResolveName.lean:367]: Just to be a bit more specific

File: src/Lean/ResolveName.lean (hunk 1 above)
Comment (nomeata) [reply src/Lean/ResolveName.lean:366]: Thanks for catching this!

File: src/Lean/ResolveName.lean [hunk 3]
```diff
@@ -320,11 +320,15 @@ def preprocessSyntaxAndResolve [Monad m] [MonadResolveName m] [MonadEnv m] [Mona
...
 /--
-Interpret the syntax `n` as an identifier for a global constant, and return a list of resolved
+Interprets the syntax `n` as an identifier for a global constant, and returns a list of resolved
 constant names that it could be referring to based on the currently open namespaces.
 This should be used instead of `resolveGlobalConstCore` for identifiers taken from syntax
 because `Syntax` objects may have names that have already been resolved.
 
+Consider using `realizeGlobalConst` if you need to handle reserved names, and
+`resolveGlobalConstWithInfo` resp. `realizeGlobalConstWithInfo` if you want the syntax to have
+hover.
```
Comment (nomeata) [inline src/Lean/ResolveName.lean:330]:
```suggestion
`realizeGlobalConstWithInfo` if you want the syntax to show the resulting name's info on hover.
```

**Pull request merged**